File: src/test/java/org/vanautrui/languages/parsing/astnodes/nonterminals/ArrayConstantNodeTest.java
package org.vanautrui.languages.parsing.astnodes.nonterminals;

import org.junit.Assert;
import org.junit.Test;
import org.vanautrui.languages.lexing.collections.TokenList;
import org.vanautrui.languages.lexing.tokens.IntegerNonNegativeConstantToken;
import org.vanautrui.languages.lexing.tokens.SymbolToken;
import org.vanautrui.languages.parsing.astnodes.nonterminal.ArrayConstantNode;

public class ArrayConstantNodeTest {

    @Test
    public void test_empty_array() throws Exception {

        TokenList list = new TokenList();
        list.add(new SymbolToken("["));
        list.add(new SymbolToken("]"));
        ArrayConstantNode arrayConstantNode = new ArrayConstantNode(list);
        Assert.assertEquals(0,arrayConstantNode.elements.size());
    }

    @Test
    public void test_simple_array() throws Exception {

        TokenList list = new TokenList();
        list.add(new SymbolToken("["));
        list.add(new IntegerNonNegativeConstantToken(1));
        list.add(new SymbolToken("]"));
        ArrayConstantNode arrayConstantNode = new ArrayConstantNode(list);
        Assert.assertEquals(1,arrayConstantNode.elements.size());
    }

    @Test
    public void test_array_multiple_elements() throws Exception {

        TokenList list = new TokenList();
        list.add(new SymbolToken("["));
        list.add(new IntegerNonNegativeConstantToken(1));
        list.add(new SymbolToken(","));
        list.add(new IntegerNonNegativeConstantToken(2));
        list.add(new SymbolToken("]"));
        ArrayConstantNode arrayConstantNode = new ArrayConstantNode(list);

        Assert.assertEquals(2,arrayConstantNode.elements.size());
    }
}

File: dragon-parser/src/test/astnodes/nonterminals/ArrayConstantNodeTest.cpp
package org.vanautrui.languages.test.astnodes.nonterminals;

import org.vanautrui.languages.compiler.lexing.tokens.IntegerNonNegativeConstantToken;
import org.vanautrui.languages.compiler.lexing.tokens.no_members.CommaToken;
import org.vanautrui.languages.compiler.lexing.tokens.no_members.LBracketToken;
import org.vanautrui.languages.compiler.lexing.tokens.no_members.RBracketToken;
import org.vanautrui.languages.compiler.lexing.utils.TokenList;
import org.vanautrui.languages.compiler.parsing.astnodes.nonterminal.ArrayConstantNode;

import java.nio.file.Paths;

public final class ArrayConstantNodeTest {

	public static int test_empty_array() throws Exception {

		final TokenList list = new TokenList(Paths.get("/dev/null"));

		list.add(new LBracketToken());
		list.add(new RBracketToken());
		final ArrayConstantNode arrayConstantNode = new ArrayConstantNode(list);

		boolean assert1 = (0 == arrayConstantNode.elements.size());

		return (assert1)?1:0;
	}

	public static int test_simple_array() throws Exception {

		final TokenList list = new TokenList(Paths.get("/dev/null"));

		list.add(new LBracketToken());
		list.add(new IntegerNonNegativeConstantToken(1));
		list.add(new RBracketToken());

		final ArrayConstantNode arrayConstantNode = new ArrayConstantNode(list);

		final boolean assert1 = (1 == arrayConstantNode.elements.size());

		return (assert1)?1:0;
	}

	public static int test_array_multiple_elements() throws Exception {

		final TokenList list = new TokenList(Paths.get("/dev/null"));

		list.add(new LBracketToken());
		list.add(new IntegerNonNegativeConstantToken(1));
		list.add(new CommaToken());
		list.add(new IntegerNonNegativeConstantToken(2));
		list.add(new RBracketToken());

		final ArrayConstantNode arrayConstantNode = new ArrayConstantNode(list);

		final boolean assert1 = (2 == arrayConstantNode.elements.size());

		return (assert1)?1:0;
	}
}
